include/bluetooth/services/cgms.h

```diff
@@ -0,0 +1,108 @@
+ */
+
+#ifndef BT_CGMS_H_
+#define BT_CGMS_H_
+
+#include <zephyr/types.h>
+#include <zephyr/bluetooth/conn.h>
+
+/**@file
+ * @defgroup bt_cgms Continuous Glucose Monitoring Service API
+ * @{
+ * @brief API for the Continuous Glucose Monitoring Service (CGMS).
+ */
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/** @brief CGMS Feature characteristic Type field */
+enum bt_cgms_feat_type {
+	/** Capillary Whole blood */
+	BT_CGMS_FEAT_TYPE_CAP_BLOOD = 0x01,
+	/** Capillary Plasma */
+	BT_CGMS_FEAT_TYPE_CAP_PLASMA = 0x02,
+	/** Venous Whole blood */
+	BT_CGMS_FEAT_TYPE_VEN_BLOOD = 0x03,
+	/** Venous Plasma */
+	BT_CGMS_FEAT_TYPE_VEN_PLASMA = 0x04,
+	/** Arterial Whole blood */
+	BT_CGMS_FEAT_TYPE_ART_BLOOD = 0x05,
+	/** Arterial Plasma */
+	BT_CGMS_FEAT_TYPE_ART_PLASMA = 0x06,
+	/** Arterial Plasma */
+	BT_CGMS_FEAT_TYPE_UNDET_BLOOD = 0x07,
+	/** Arterial Plasma */
+	BT_CGMS_FEAT_TYPE_UNDET_PLASMA = 0x08,
+	/** Interstitial Fluid (ISF) */
+	BT_CGMS_FEAT_TYPE_FLUID = 0x09,
+	/** Control Solution */
+	BT_CGMS_FEAT_TYPE_CONTROL = 0x0A,
+};
+
+/** @brief CGMS Feature characteristic Sample Location field */
+enum bt_cgms_feat_loc {
+	/** Finger */
+	BT_CGMS_FEAT_LOC_FINGER = 0x01,
+	/** Alternate Site Test (AST) */
+	BT_CGMS_FEAT_LOC_AST = 0x02,
+	/** Earlobe */
+	BT_CGMS_FEAT_LOC_EAR = 0x03,
+	/** Control solution */
+	BT_CGMS_FEAT_LOC_CONTROL = 0x04,
+	/** Subcutaneous tissue */
+	BT_CGMS_FEAT_LOC_SUB_TISSUE = 0x05,
+	/** Sample Location value not available */
+	BT_CGMS_FEAT_LOC_NOT_AVAIL = 0x0F,
+};
+
+/** @brief Continuous Glucose Monitoring service initialization structure */
+struct bt_cgms_init_param {
+	/** Continuous Glucose Monitoring type */
+	enum bt_cgms_feat_type type;
+	/** CGM sample location */
+	enum bt_cgms_feat_loc sample_location;
+	/** The session run time in uint of hour */
+	uint16_t session_run_time;
+	/** Initial communication interval of notification */
+	uint16_t initial_comm_interval;
+	/** The shortest communication interval allowed */
+	uint16_t shortest_comm_interval;
+};
+
+/** @brief Initialize Continuous Glucose Monitoring service.
+ *
+ * This will initialize components used in CGMS.
+ *
+ *  @param[in] init_params The parameter used to initialize the corresponding
+ *             values of CGMS monule.
+ *
+ *  @return Zero in case of success and error code in case of error.
+ */
+int bt_cgms_init(struct bt_cgms_init_param *init_params);
+
+/** @brief Submit glucose concentration measurement to CGM service.
+ *
+ * This will add the new glucose measurement into the database.
+ *
+ *  @param[in] glucose The glucose concentration measurement in milligram per deciliter (mg/dL).
+ *             It is in 16-bit SFLOAT type, comprising 4-bit exponent and signed 12-bit mantissa.
+ *
+ *  @return Zero in case of success and error code in case of error. If it is -EBUSY, it means
+ *             the database is busy, and the application may try to add the record later.
+ */
+int bt_cgms_measurement_add(uint16_t glucose);
```

Comment:
Consider using a structure here like in the nRF5 SDK API:

```suggestion
int bt_cgms_measurement_add(struct bt_cgms_meas measurement);
```

```
struct bt_cgms_meas {
        /** Glucose concentration. 16-bit word comprising 4-bit exponent and signed 12-bit mantissa. */
	uint16_t glucose_concentration;
};
```

If it is required to have more configuration parameters for measurement in the future, then we will add them as structure fields without breaking existing APIs (e.g. Quality of the measurement, Trend information, etc.).